### DISCUSSION TITLE
Pipelined Processor Diagram lacks ALUSrcA

### DISCUSSION THREAD

**Discussion by kxing28** [NUS-CG3207/labs#25] [Lab 4] [answered]
**Pipelined Processor (Ch 5 Slide 11)**
![image](https://github.com/user-attachments/assets/b09965cb-b351-481c-9598-9fccbc2e8687)

**Single-Cycle Processor (Ch 3b Slide 23)**
![image](https://github.com/user-attachments/assets/b58d0f97-3036-4b3b-b3a9-cf66a6342446)

In comparing the 2 processors, it seems that the Pipelined processor has lost `ALUSrcA`. My partner and I can't wrap our heads around this -- `luipc` and `jal` both require the ability to write a PC to register, which is lost when ALUSrcA goes away. We also can't do `lui` if we lose `ALUSrcA`.  We've decided to implement the processor with ALUSrcA anyways. Can any staff confirm that this is an omission in the diagram?

**Comment by r-panicker** [comment, marked as answer]
The entire chapter on pipelining uses the most basic design as a starting point. This is good enough to illustrate the concepts of pipelining and hazard resolution, without the clutter caused by the full design. Extending the idea of pipelining to full design is very straightforward, and that is your job for Lab 4 :)
